### DISCUSSION TITLE
How to add custom account types?

### DISCUSSION THREAD

**Discussion by chiefbiiko** [MetaMask/snaps#2949] [Q&A] [answered]
Been trying to create a custom account type, so that those accounts appear in Flask.

Forked `metamask/accounts` and added the relevant account types in:
- `keyring-api/src/myapp/index.ts`
- `keyring-api/src/myapp/types.ts`
- `keyring-api/src/api/account.ts`
- `keyring-api/src/index.ts`
- `keyring-internal-api/src/types.ts`

Still when trying to create such a custom account it throws:

```
MetaMask - RPC Error: At path: params.account.type -- Expected one of `"eip155:eoa","eip155:erc4337","bip122:p2wpkh","solana:data-account"`, but received: "myapp:v0"
```

What am I missing? Do custom account types require any changes on the Flask side, like it needs the forked `keyring-internal-api`?

**Comment by danroc** [comment, marked as answer]
Custom account types are not supported yet. Can say a bit more why you want to add a new account type and why the existing ones are not enough?

**Reply by chiefbiiko** [reply]
We are building a snap to support shielded transactions. Its a whole different key pair and having a custom account type would allow for deeper integration.
